apps/dashboard/app/helpers/system_status_helper.rb

```diff
@@ -0,0 +1,105 @@
+# frozen_string_literal: true
+
+# helper for the system status page
+module SystemStatusHelper
+  # Used to not repeatedly request the job adapter
+  def update_cluster(cluster)
+    @job_adapter = cluster.job_adapter
+    @cluster_info = @job_adapter.cluster_info
+    @cluster = cluster
+  end
+
+  def job_adapter(cluster)
+    if @cluster != cluster
+      update_cluster cluster
+    end
+
+    @job_adapter
+  end
+
+  def cluster_info(cluster)
+    if @cluster != cluster
+      update_cluster cluster
+    end
+
+    @cluster_info
+  end
+
+  def title(cluster)
+    "#{cluster.metadata.title.titleize} Cluster Status"
+  end
+
+  def generic_pct(value, total)
+    "#{(value.to_f / total * 100).round 2}%"
+  end
+
+  def generic_status(active, total, name)
+    "#{active} of #{total} #{name} Active (#{total - active} Free)"
+  end
+
+  def node_status(cluster)
+    c_info = cluster_info cluster
```

Comment:
Can we use parentheses on method calls? I know Ruby allows this, but I feel like it's more readable with the enclosing ().

```ruby
 c_info = cluster_info(cluster)
```